addons/mail/models/mail_followers.py

```diff
@@ -116,39 +116,57 @@ def _get_recipient_data(self, records, subtype_id, pids=None, cids=None):
           user groups of partner (void as irrelevant if channel ID),
         """
         if records and subtype_id:
+            query_partners = """
+    SELECT partner.id as pid, cast(NULL as INTEGER) AS cid,
+            partner.active as active, partner.partner_share as pshare, NULL as ctype,
+            users.notification_type AS notif, array_agg(groups.id) AS groups
+        FROM res_partner partner
+        LEFT JOIN res_users users ON users.partner_id = partner.id AND users.active
+        LEFT JOIN res_groups_users_rel groups_rel ON groups_rel.uid = users.id
+        LEFT JOIN res_groups groups ON groups.id = groups_rel.gid
```

Comment:
this join is useless, replace `array_agg(groups.id)` with `array_agg(groups_rel.gid)` and you're on.